File: src/main/java/com/example/demoimdb/model/Comment.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import javax.security.auth.Subject;

@Entity
@Table(name = "comment")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Comment {
    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
    @ManyToOne
    @JoinColumn(name = "episode_id")
    @JsonIgnore
    private Episode episode;
    @ManyToOne
    @JoinColumn(name = "account_id")
    private Account account;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "content", length = 1000)
    private String content;
    @Column(name = "date")
    private String date;
}

File: src/entities/Comment.ts
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Movie } from './Movie';
import { Episode } from './Episode';
import { Account } from './Account';

@Entity({ name: 'comments' })
export class Comment {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ nullable: false })
  content: string;

  @Column({ nullable: false })
  date: string;

  @ManyToOne(() => Movie, (movie) => movie.comments, {
    onDelete: 'SET NULL',
  })
  movie: Movie;

  @ManyToOne(() => Episode, (episode) => episode.comments, {
    onDelete: 'SET NULL',
  })
  episode: Episode;

  @ManyToOne(() => Account, (account) => account.comments)
  account: Account;
}
